substrate/client/consensus/beefy/src/communication/gossip.rs

```diff
@@ -72,24 +70,27 @@ enum Consider {
 
 /// BEEFY gossip message type that gets encoded and sent on the network.
 #[derive(Debug, Encode, Decode)]
-pub(crate) enum GossipMessage<B: Block> {
+pub(crate) enum GossipMessage<B: Block, AuthorityId: AuthorityIdBound> {
 	/// BEEFY message with commitment and single signature.
-	Vote(VoteMessage<NumberFor<B>, AuthorityId, Signature>),
```

Comment:
I just drop this here, but applies all over the place.
Does `<AuthorityId as AppCrypto>::Signature` work? (in place of `as RuntimeAppPublic`.
I can't try it right now, but if yes then to me looks more appropriate using a trait that has not being designed for usage within the runtime

Reply:
We had this conversation somewhere else before, I'm sorry my PR takes ages to merge :-/ `AppCrypto::Signature` somehow is not bounded by `Codec` (perhaps it should), `RuntimeAppPublic::Signature` has `Codec` and I assume you need to encode and send the Vote. I can check nonetheless.

Reply:
yep!:

```
error[E0599]: the variant or associated item `decode_all` exists for enum `GossipMessage<B, AuthorityId>`, but its trait bounds were not satisfied
   --> substrate/client/consensus/beefy/src/worker.rs:862:40
    |
862 |                         GossipMessage::<B, AuthorityId>::decode_all(&mut &notification.message[..])
    |                                                          ^^^^^^^^^^ variant or associated item cannot be called on `GossipMessage<B, AuthorityId>` due to unsatisfied trait bounds
    |
   ::: substrate/client/consensus/beefy/src/communication/gossip.rs:73:1
    |
73  | pub(crate) enum GossipMessage<B: Block, AuthorityId: AuthorityIdBound> {
```

Reply:
So I guess that we can merge this as soon as CI passes